iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java

```diff
@@ -77,7 +106,61 @@ public void processPushSubscriptionEvent(PushSubscriptionEvent event) throws IOE
         }
     }
 
-    void updateUser(Event.Context context) throws IOException {
+    void updateUser(EventProcessingRequest request) throws IOException {
+        Event.Context context = new Event.Context(request);
+
+        if (request.getEvents() != null) {
+            List<UserIdentityChangeEvent> emailChangeEvents = request.getEvents().stream()
+                    .filter(e -> e.getType() == Event.Type.USER_IDENTITY_CHANGE)
+                    .map(e -> (UserIdentityChangeEvent) e)
+                    .filter(e -> e.getAdded() != null && e.getRemoved() != null)
+                    .filter(e -> e.getAdded().size() > 0 && e.getRemoved().size() > 0)
+                    .filter(e -> e.getAdded().get(0).getType().equals(UserIdentity.Type.EMAIL))
+                    .filter(e -> !isEmpty(e.getAdded().get(0).getValue()) && !isEmpty(e.getRemoved().get(0).getValue()))
+                    .collect(Collectors.toList());
+
+            List<UserIdentityChangeEvent> emailAddedEvents = request.getEvents().stream()
+                    .filter(e -> e.getType() == Event.Type.USER_IDENTITY_CHANGE)
+                    .map(e -> (UserIdentityChangeEvent) e)
```

Comment:
Another minor nit pick. Could this be refactored:

    request.getEvents().stream()
    .filter(e -> e.getType() == Event.Type.USER_IDENTITY_CHANGE)
    .map(e -> (UserIdentityChangeEvent) e)```

? (Not sure if one can re-use filtered results like that)